convert2rhel/checks.py

```diff
@@ -16,13 +16,194 @@
+        if distinct_kernel_pkgs[0].startswith(("kernel", "kmod"))
+    )
+
+
+def get_rhel_supported_kmods(_refresh_system_info=False):
+    """Return set of target RHEL supported kernel modules."""
+    # allowing to rerun resolve_system_info (for the needs of tests)
+    if _refresh_system_info:
+        system_info.resolve_system_info()
+    repoquery_repoids_args = " ".join(
+        "--repoid " + repoid for repoid in system_info.default_rhsm_repoids
+    )
+    # get output of a command to get all packages which are the source
+    # of kmods
+    kmod_pkgs_bytes, _ = run_subprocess(
+        (
+            "repoquery "
+            "--releasever={releasever} "
+            "{repoids_args} "
+            "-f /lib/modules/*.ko*"
+        ).format(
+            releasever=system_info.releasever,
+            repoids_args=repoquery_repoids_args,
+        )
+    )
+    # from these packages we select only the latest one
+    kmod_pkgs = get_most_recent_unique_kernel_pkgs(
+        kmod_pkgs_bytes.decode().rstrip("\n").split()
+    )
+    rhel_kmods = set()
+    for kmod_pkg in kmod_pkgs:
+        # querying obtained packages for files they produces
+        rhel_kmods_bytes, _ = run_subprocess(
+            (
+                "repoquery "
+                "--releasever={releasever} "
+                "{repoids_args} "
+                "-l {pkg}"
+            ).format(
+                releasever=system_info.releasever,
+                repoids_args=repoquery_repoids_args,
+                pkg=kmod_pkg,
+            )
+        )
+        # appending only files which are kmods (ends with ko or ko.xz)
+        rhel_kmods.update(
+            set(
+                _get_kmod_comparison_key(kmod_path)
+                for kmod_path in filter(
+                    lambda path: path.endswith(("ko.xz", "ko")),
+                    rhel_kmods_bytes.decode().rstrip("\n").split(),
+                )
+            )
+        )
+    return rhel_kmods
+
+
+def check_thirdparty_kmods():
+    unsigned_modules, _ = run_subprocess("grep '(' /proc/modules")
+    if unsigned_modules:
+        logger.critical(
+            "Unsigned kernel module(s) has been detected. "
+            "Third-party components are not supported per our "
+            "software support policy\n%s.\n\n "
+            "Uninstall or disable the module(s) and run convert2rhel "
+            "again to continue with the conversion.",
+            LINK_KMODS_RH_POLICY,
+        )
+
+
+def ensure_compatibility_of_kmods(_refresh_system_info=False):
+    """Ensure if the host kernel modules are compatible with RHEL."""
+    check_thirdparty_kmods()
+    host_kmods = get_host_kmods()
+    rhel_supported_kmods = get_rhel_supported_kmods(_refresh_system_info)
+    if not host_kmods.issubset(rhel_supported_kmods):
+        logger.critical(
+            "The following kernel module(s) are not supported "
+            "in RHEL:\n %s" % "\n".join(host_kmods - rhel_supported_kmods)
```

Comment:
This currently prints path relative to `/lib/modules/$(uname -r)/` which can be confusing, we should print absolute path here.

Reply:
added